lib/external_config/s3.ex

```diff
@@ -30,4 +30,40 @@ defmodule ExternalConfig.S3 do
         {nil, %{}}
     end
   end
+
+  def get_active_headend_ip() do
+    s3_client = Application.get_env(:realtime_signs, :s3_client)
+    aws_client = Application.get_env(:realtime_signs, :aws_client)
+    bucket = Application.get_env(:realtime_signs, :s3_bucket)
+    path = Application.get_env(:realtime_signs, :s3_active_headend_path)
+
+    case s3_client.get_object(bucket, path)
+         |> aws_client.request() do
+      {:ok, response} ->
+        body = Jason.decode!(response.body)
+        {:ok, body["active_headend_ip"]}
+
+      {:error, e} ->
+        Logger.error("active_headend_ip: s3 response error: #{inspect(e)}")
```

Comment:
nit: Since we're already logging errors at the call site, we could just return the error here and have that one print it out.